process/block/baseProcess.go

```
@@ -3318,13 +3319,15 @@ func (bp *baseProcessor) verifyGasLimit(header data.HeaderHandler, miniBlocks bl
 
 	bp.gasComputation.Reset()
 	_, numPendingMiniBlocks, err := bp.gasComputation.AddIncomingMiniBlocks(splitRes.incomingMiniBlocks, splitRes.incomingTransactions)
-	if err != nil {
+	// if limits are reached, do not propagate the error so consensus can continue but with no blocks
```

I think on the zero limit condition we should instead not do any selection for incoming and outgoing miniblocks/transactions, then we should not need to ignore this error.

The block production should continue, while there is a zero limit, but with no transactions.

---

fixed